src/core/tracking.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable no-underscore-dangle */
+
+import config from 'config';
+import log from 'core/logger';
+
+
+export class Tracking {
+
+  constructor(opts = {}) {
+    this.id = opts.trackingId;
+    this.enabled = opts.enabled;
+    this.logPrefix = `[GA: ${this.enabled ? 'ON' : 'OFF'}]`;
+    this.initialized = false;
+  }
+
+  _ga(args) {
+    if (!this.initialized) {
+      throw new Error('Must call init() first');
```

Comment:
Hmm - looking at the docs, I reckon it would be better to refactor this to not have init at all and use this instead: https://developers.google.com/analytics/devguides/collection/analyticsjs/#alternative_async_tracking_snippet